src/stdatamodels/jwst/datamodels/tests/test_read_metadata.py

```diff
@@ -0,0 +1,91 @@
+"""Test utilities for loading metadata without loading entire model."""
+
+import pytest
+import numpy as np
+from sys import getsizeof
+from astropy.time import Time
+import stdatamodels.jwst.datamodels as dm
+from stdatamodels.jwst.datamodels.util import read_metadata
+
+
+TESTFILE_ROOT = "jwst_image."
+FILT = "GR150R"
+INPUT_TIME = "2021-01-01 00:00:00.000"
+DATA = np.zeros((1000, 1000), dtype=np.float32)
+
+
+@pytest.fixture
+def make_and_save_models(tmp_path):
```

Comment:
```suggestion
@pytest.fixture
def make_and_save_models(tmp_path):
```
This could be a parametrized fixture which when used would paramerized the tests that use it. Something like:
```python
@pytest.fixture(params=["fits", "asdf"])
def saved_model_filename(request, fake_model,tmp_path):
    ....
    extension = request.param
```
Also if it returned the path `tmp_path` wouldn't need to be added to the tests that use this fixture.
Then `test_read_metadata` could be something like:
```python
def test_read_metadata(saved_model_filename):
```
without the need for parametrization or the other fixtures.